### DISCUSSION TITLE
Feature request: Similar methods, different parameters

### DISCUSSION THREAD

**Discussion by lamazze** [sebastienros/jint#1558] [Ideas]
Feature request.

Given this c# class:
```cs
public class TestClass
{
        public string Value { get; set; }

        public TestClass this[string Address]
        {
            get { Value = "ByAdress"; return this; }
        }

        public TestClass this[int Row, int Col]
        {
            get { Value = "ByArray2"; return this; }
        }

        public TestClass this[int fRow, int fCol, int tRow, int tCol]
        {
            get { Value = "ByArray4"; return this; }
        }
}
```
And this code:

```cs
TestClass Test = new TestClass();

var engine = new Engine().SetValue("log", new Action<object>(Console.WriteLine));
engine.SetValue("Test", Test);

engine.Execute("log(Test[\"A\"].Value);");
```
With Jint latest stable: you get ByArray not matter what call you Test["A"], Test[0, 0] or Test[0, 0, 0, 0] or in fact anything between the []
With Jint latest beta: you get nothing meaningful not matter what call Test

It should return ByAddress (Test["A"]), ByArray2 (Test[0, 0]), ByArray4 (Test[0, 0, 0, 0].

**Comment by lahma** [comment]
Can you help out with the PR?

**Comment by lamazze** [comment]
Hello,
I am a really low skilled programmer so I’m afraid I can’t.
I’ve created a project with jint source and built that little example above. 
I’ve tried to debug the code but got lost in all the jumps and in fact can’t understand how and where ‘the magic happens’.
